STM32F405RGTx/Core/Inc/state_machine_thread.hpp

```diff
@@ -0,0 +1,109 @@
+/* state_machine.hpp
+ *
+ *    Created on: Jul. 11, 2021
+ *            Author: Kanishk Kashyap
+*/
+#pragma once
+
+#include <stdint.h>
+#include "can.h"
+#include "cmsis_os.h"
+#include "can.h"
+#include "util.hpp"
+
+// Constants - These numbers are arbritrary should be changed after testing
+// Mosfet Array Length
+#define NUM_MOSFETS                 6
+#define NUM_PHASE_OUTPUTS           3
+
+// Severe Fault
+#define MAX_VOLTAGE_SEVERE          4.0F
+#define MIN_VOLTAGE_SEVERE          1.0F
+#define MAX_CURRENT_SEVERE          50.0F
+#define MAX_DC_VOLTAGE_SEVERE       4.0F
+#define MIN_DC_VOLTAGE_SEVERE       1.8F
+#define MAX_FET_TEMP_SEVERE         70.0F
+
+// Normal Fault
+#define MAX_VOLTAGE_NORMAL          3.8F
+#define MIN_VOLTAGE_NORMAL          1.8F
+#define MAX_CURRENT_NORMAL          40.0F
+#define MAX_DC_VOLTAGE_NORMAL       4.0F
+#define MIN_DC_VOLTAGE_NORMAL       1.8F
+#define MAX_FET_TEMP_NORMAL         60.0F
+
+// DC_CAPTEMP
+#define MAX_DCCAP_TEMP_NORMAL       5
+#define MAX_DCCAP_TEMP_SEVERE       6
+
+// MISC
+#define DECCELERATION_DISTANCE      10 // turn off motor at 10m before the end of the track
```

Comment:
Typo -> this should be corrected this to `DECELERATION_DISTANCE` wherever the define is used. Could lead to confusion down the line